### ISSUE TITLE
[Bug]: [restful v2] search api doesn't work well with searchParams

### ISSUE THREAD

**Issue by wwli05** [milvus-io/milvus#34105]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.3.17
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): restful v2
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1.in code [internal/distributed/proxy/httpserver/handler_v2.go], SearchReqV2 struct contains filed [params] , but in restful v2
tutorial section it declares [searchParams] which does not seem match;
2.if i want to declare special param just like [nprobe](FOR IVF_FLAT index) in [params], it does not take effect when i follows the next code in [handler_v2.go]

### Expected Behavior

when I declare special search param, it takes effect so that I can take controll of various query params.

**Comment by yanliang567** [comment]
please try the code snippet below, or share some code for reproduction
```
search_params = {
    "metric_type": "IP",
    "params": {"nprob": 128}
}

# Search with limit
res = client.search(
    collection_name="test_collection",
    data=data,
    limit=3,
    search_params=search_params
)
```

/assign @wwli05 
/unassign

**Comment by wwli05** [comment]
pls check below code
curl --location --request POST "http://${}/v2/vectordb/entities/search" \
--header "" \
--data-raw '{
   "collectionName":"",
   "data": [],
  "annsField": "vector",
  "searchParams": [
      "metricType":"IP",
         "params":{
               "nprobe":16  ---we adjust this value, but the performance has no impact, it seems this param does not take effect in Rest V2 
  ]
}'

**Comment by xiaofan-luan** [comment]
/assign @PowderLi 
can you check on that?

**Comment by wwli05** [comment]
@yanliang567 could you confirm and fix it ? thanks

**Comment by yanliang567** [comment]
/assign @smellthemoon 
please help to double check the issue

**Comment by ljpassingby** [comment]
Is this problem solved in milvus-2.5.5?
Is there any demo about nprobe param for RESTful api which makes sense?
